### PR TITLE
Remove f16, bf16 from node's evaluate methods v2

### PR DESCRIPTION
**Pull request by mateusztabaka**
Since this change, most of operation that override Node::evaluate will stop instantiating evaluate methods for f16 and bf16. There are few exceptions - we still keep f16 (and/or bf16) evaluates for the following operations:
- Ceiling
- Convert
- FakeConvert

Primary reason for that is to reduce binary size. The change saves us around 200 KB. The change is transparent to the caller, so you can still evaluate f16/bf16 operations, but internally they'll be executed with f32 precision.

Ticket: CVS-108489

### CODE REVIEW

File: src/core/src/op/subtract.cpp [hunk 1]
```diff
@@ -48,16 +48,19 @@ bool Subtract::evaluate(TensorVector& outputs, const TensorVector& inputs) const
...
-                      inputs[1],
-                      outputs[0],
-                      inputs[0].get_shape(),
-                      inputs[1].get_shape(),
-                      get_autob());
+    return IF_TYPE_OF_CONVERT_TENSORS(v1_Subtract_evaluate,
+                                      this,
+                                      outputs,
+                                      inputs,
+                                      OV_PP_ET_LIST(f32, i8, i32, i64, u8, u32, u64),
```
[the remark this answers is not in the document]
Comment (mateusztabaka) [reply src/core/src/op/subtract.cpp:55]: Even if the list is reduced - this operation's evaluate can still handle f16, bf16.

**Review by praasz — APPROVED** [review]

File: src/core/src/node.cpp [hunk 2]
```diff
@@ -750,12 +750,13 @@ bool ov::Node::visit_attributes(AttributeVisitor&) {
 
 namespace ov {
 void check_new_args_count(const Node* const node, const OutputVector& new_args) {
+    size_t node_input_size = node->get_input_size();
     NODE_VALIDATION_CHECK(node,
-                          new_args.size() == node->input_values().size(),
+                          new_args.size() == node_input_size || node_input_size == 0,
```
[the remark this answers is not in the document]
Comment (mateusztabaka) [reply src/core/src/node.cpp]: actually the change may not be necessary

**Review by itikhono — APPROVED** [review]

**Review by rkazants — COMMENTED** [review]
What about reduction at the cost of removing i32 and use i64 for it? And u32->u64?

File: src/core/src/constant_fold_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+#include "openvino/op/range.hpp"
+#include "openvino/op/util/assign_base.hpp"
+#include "openvino/op/util/multi_subgraph_base.hpp"
+#include "openvino/op/util/read_value_base.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include "openvino/reference/convert.hpp"
+#include "ov_ops/type_relaxed.hpp"
+
+const ov::element::TypeVector& ov::util::unsupported_types() {
+    static const ov::element::TypeVector types{ov::element::f16, ov::element::bf16};
```
[the remark this answers is not in the document]
Comment (praasz) [reply src/core/src/constant_fold_utils.cpp:19]: The goal was to remove the f16, and bf16 from core ops to reduce bin-size but  constant folding and shape inference (before apply the plugin conversion) still requires these precisions to do calculations and this is reason to apply convert to f32 without changing the model. In core when operator do calculations on f16, bf16 values are converted to float and then back (element by element)

The i64 and i32 are common types when used for shapes calculations (constant fold and shape inference) and apply conversion will may introduce data copies native support I think is better.

**Pull request merged**